staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -108,6 +108,14 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 		userInfo, _ := request.UserFrom(ctx)
 		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		admissionCheck := func(updatedObject runtime.Object, currentObject runtime.Object) error {
+			isNotZeroObject, err := hasUID(currentObject)
```

Comment:
`isZeroObject := !hasUID(...)` makes the logic simpler

Reply:
@sttts I'm afraid we cannot do this because unary operator `!` can't be applied to a function returning 2 values.🧐

Reply:
add another helper. Or even simpler: isPreExistingObject instead of isNotZeroObject. Just to get rid of the double negation.